lib/k8s/client.ex

```diff
@@ -436,37 +439,40 @@ defmodule K8s.Client do
         api_version: "apps/v1",
         name: "Deployment",
         path_params: [namespace: "test", name: "nginx"],
-        data: K8s.Resource.from_file!("test/support/manifests/nginx-deployment.yaml")
+        data: K8s.Resource.from_file!("test/support/manifests/nginx-deployment.yaml"),
+        header_params: ["Content-Type": "application/merge-patch+json"]
       }
   """
   @spec patch(map()) :: Operation.t()
-  def patch(%{} = resource), do: Operation.build(:patch, resource)
+  def patch(%{} = resource), do: Operation.build(:patch, resource, patch_type: :merge)
```

Comment:
Also calling out, that there is an api inconsistency here, in that I was not able to add `patch_type \\ :merge` for this function since it conflicts with `patch/2`

Reply:
Fixed with b5cb5716